Fallback for TaskType

stiegerb opened the pull request:
- TaskType is now set to 'UNKNOWN' in case it can't be determined in the hardcoded logic
- I also refactored the logic into a separate function, together with `Campaign` and `Workflow`

This will reduce TaskType to be one the following values:
`Cleanup`, `Merge`, `LogCollect`, `MINIAOD`, `DIGIRECO`, `MINIAOD`, `GENSIM`, `DIGI`, `RECO`, `GENSIM`, `UNKNOWN`

We should probably also take this opportunity to update the logic itself.

I'm not fully happy with this as there seem to be a lot of useful values that we'd lose to `UNKNOWN` now, e.g. `DataProcessing`, `Reco`, `Repack`, `Express`. Do we want to keep these? @bbockelm @amaltaro @belforte 

For a full list of the possible values now, see here:
https://es-cms.cern.ch/kibana/goto/59ae8ce642832bf4a217ca6d3631bd13

amaltaro commented:
Yes, we should keep those and probably a few others.

I'd actually make a list of knowTaskTypes (or goodTaskTypes) such that we can use that list and skip the regex guessing the task type. Something like:
```
if ad.get('CMS_JobType', '') in ("Merge", "Harvesting", "Cleanup", "LogCollect", "Skim", "Express", "Repack"):
    return ad.get('CMS_JobType', '')
else:
    return guessTaskType...
```

I didn't go through the whole list of tasks, but `DataProcessing` is (or could be) mapped to ReReco.

bbockelm commented:
I would prefer to keep `DataProcessing`, `Reco`, `Repack`, `Express`.  We are mostly aiming to get rid of having an infinite number of values -- adding a few more enumerated ones is OK.

stiegerb commented:
Ok, I added a commit with those suggestions.

@bbockelm Do we want to clear up the difference between `Reco` and `RECO`?

I also changed the default from `UNKNOWN` to `Unknown`, to be consistent with `CMS_JobType`.

stiegerb commented:
Ok, after consulting with Alan, we decided to merge "Reco" and "RECO" (and in fact all capitalization variations of "reco"), and call them all "RECO".

I'll go ahead and merge this so we can move forward.
